src/traces/bar/index.js

```diff
@@ -9,38 +9,29 @@
 
 'use strict';
 
-var Plotly = require('../../plotly');
-
-Plotly.Plots.register(exports, 'bar',
-    ['cartesian', 'bar', 'oriented', 'markerColorscale', 'errorBarsOK', 'showLegend'], {
-        description: [
-            'The data visualized by the span of the bars is set in `y`',
-            'if `orientation` is set th *v* (the default)',
-            'and the labels are set in `x`.',
-
-            'By setting `orientation` to *h*, the roles are interchanged.'
-        ].join(' ')
-    }
-);
-
-exports.attributes = require('./attributes');
-
-exports.layoutAttributes = require('./layout_attributes');
-
-exports.supplyDefaults = require('./defaults');
-
-exports.supplyLayoutDefaults = require('./layout_defaults');
-
-exports.calc = require('./calc');
-
-exports.setPositions = require('./set_positions');
-
-exports.colorbar = require('../scatter/colorbar');
-
-exports.arraysToCalcdata = require('./arrays_to_calcdata');
-
-exports.plot = require('./plot');
-
-exports.style = require('./style');
-
-exports.hoverPoints = require('./hover');
+var Bar = {};
+
+Bar.attributes = require('./attributes');
+Bar.layoutAttributes = require('./layout_attributes');
+Bar.supplyDefaults = require('./defaults');
+Bar.supplyLayoutDefaults = require('./layout_defaults');
+Bar.calc = require('./calc');
+Bar.setPositions = require('./set_positions');
+Bar.colorbar = require('../scatter/colorbar');
+Bar.arraysToCalcdata = require('./arrays_to_calcdata');
+Bar.plot = require('./plot');
+Bar.style = require('./style');
+Bar.hoverPoints = require('./hover');
+
+Bar._categories = ['cartesian', 'bar', 'oriented', 'markerColorscale', 'errorBarsOK', 'showLegend'];
+Bar._type = 'bar';
+Bar._meta = {
```

Comment:
Good point. I used the `_` mainly because they're static properties that should only really be used by `register`, whereas the other parts vary from module to module and get used elsewhere, but we could go either way. Up to you.